examples/step-61/step-61.cc

```diff
@@ -289,138 +315,178 @@ namespace Step61
+  // define a Raviart-Thomas finite element object, and have FEValues
+  // objects that evaluate it at quadrature points. We then need to
+  // compute the matrix $C^K$ on every cell $K$, for which we need the
+  // matrices $M^K$ and $G^K$ mentioned in the introduction.
+  //
+  // A point that may not be obvious is that in all previous tutorial
+  // programs, we have always called FEValues::reinit() with a cell
+  // iterator from a DoFHandler. This is so that one can call
+  // functions such as FEValuesBase::get_function_values() that
+  // extract the values of a finite element function (represented by a
+  // vector of DoF values) on the quadrature points of a cell. For
+  // this operation to work, one needs to know which vector elements
+  // correspond to the degrees of freedom on a given cell -- i.e.,
+  // exactly the kind of information and operation provided by the
+  // DoFHandler class.
+  //
+  // On the other hand, we don't have such a DoFHandler object for the
+  // Raviart-Thomas space in this program. In fact, we don't even have
+  // an element that can represent the "broken" Raviart-Thomas space
+  // we really want to use here (i.e., the restriction of the
+  // Raviart-Thomas shape functions to individual cells, without the
+  // need for any kind of continuity across cell interfaces. We solve
+  // this conundrum by using the fact that one can call
+  // FEValues::reinit() also with cell iterators into Triangulation
+  // objects (rather than DoFHandler objects). In this case, FEValues
+  // can of course only provide us with information that only
+  // references information of cells, rather than degrees of freedom
+  // enumerated on these cells. So we can't use
+  // FEValuesBase::get_function_values(), but we can use
+  // FEValues::shape_value() to obtain the values of shape functions
+  // at quadrature points on the current cell. It is this kind of
+  // functionality we will make use of below.
+  //
+  // Given this introduction, the following declarations should be
+  // pretty obvious:
   template <int dim>
   void WGDarcyEquation<dim>::assemble_system()
   {
     const FE_RaviartThomas<dim> fe_rt(0);
-    const QGauss<dim>           quadrature_formula(fe_rt.degree + 1);
-    const QGauss<dim - 1>       face_quadrature_formula(fe_rt.degree + 1);
-    const RightHandSide<dim>    right_hand_side;
-
-    // We define objects to evaluate values and
-    // gradients of shape functions at the quadrature points.
-    // Since we need shape functions and normal vectors on faces, we need
-    // <code>FEFaceValues</code>.
-    FEValues<dim> fe_values_rt(fe_rt,
-                               quadrature_formula,
-                               update_values | update_gradients |
-                                 update_quadrature_points | update_JxW_values);
 
-    FEValues<dim> fe_values(fe,
+    const QGauss<dim>     quadrature_formula(fe_rt.degree + 1);
```

Comment:
Given the fact that you use the generic way `fe_rt.degree+1` here but use the hard-coded `const FE_RaviartThomas<dim> fe_rt(0);` in line 366 above, I wonder whether you would like to switch that line to `const FE_RaviartThomas<dim> fe_rt(fe.degree);` as well. I'm not familiar with all details of the program but would assume that this is the proper generalization in case the finite element in the main problem class' constructor is changed to a higher order element.

Reply:
I'd like to leave this to a later PR if that's ok with you. What it boils down to is that it's likely useful to couple the degree of the RT element to the degree of the other elements, but I don't know for sure. I'll open an issue for this elsewhere, but I think it's not necessary to address this for the upcoming release.